sherpa/astro/fake.py

```diff
@@ -21,111 +21,180 @@
+       pileup_model, add_bkgs, bkg_models, and id arguments are
+       no-longer used. The include_bkg_data and method arguments
+       have been added.
 
     Parameters
     ----------
     data : sherpa.astro.data.DataPHA
         The dataset (may be a background dataset).
-    model : sherpa.models.model.ArithmeticModel instance
-        The model that will be used for simulations.
-    is_source : bool
-        ``True`` means that the ``model`` does not contain response or
-        background components and that these need to be added based on the
-        ARF, RMF, and backgrounds set up for the data set. If ``False``, then
-        the ``model`` contains any components to describe the instrument
-        already.
-    pileup_model : None or `sherpa.astro.models.JDPileup` instance
-        Pileup Model for the source spectrum
-    add_bkgs : bool
-        If ``True`` backgrounds are added to the source counts.
-    bkg_srcs : dict
-        Keys in the dictionary need to be the background ids in the dataset
-        ``data``, and the values are the corresponsing source models. For all
-        background datasets that are listed in this dictionary, the
-        background counts will be simulated based on the model, appropriately
-        scaled (for area etc.) and added to the source. The same ``is_source``
-        setting as for the source model applies here, i.e. if the source model
-        already contains the ARF and the RMF, then the background models should
-        do so, too. This setting has no effect if ``add_bkgs=False``.
-
-        For all background ids not listed in this dictionary, the
-        counts will be drawn from the PHA data of the background data set.
-    id : str
-        String with id number if called from UI layer. This is only
-        used for certain error messages.
+    model : sherpa.models.model.ArithmeticModel
+        The model that will be used for simulations. It must contain
+        any background components, appropriately scaled, and include
+        the relevant response.
+    include_bkg_data : bool, optional
+        Should the counts in the background datasets be included when
+        calculating the predicted signal? As background datasets are
+        often noisy it is generally better to include a model of the
+        background in the model argument.
     method : callable or None
         The routine used to simulate the data. If None (the default)
-        then sherpa.utils.poisson_noise is used, otherwise the function
-        must accept a single ndarray, returning a ndarray of the same shape.
+        then sherpa.utils.poisson_noise is used, otherwise the
+        function must accept a single ndarray, returning a ndarray of
+        the same shape.
+
+    Notes
+    -----
+
+    The model must include the relevant response, and if a background
+    model is required then it must also be included. The simulation
+    requires an array of "predicted counts" (one for each channel).
```

Comment:
"....requires an array of predicted counts..." - this is confusing, as the simulations generate an array of predicted counts based on the evaluated model as stated in the next sentence. It is not required as input.

Reply:
Technically I'd argue that the text isn't talking about the input argument here, but I'll see what I can do to clean the text up.